### PR TITLE
feat(tracing): deprecate use of B3 propagator class from API package - use Extensions.Propagators package instead

### PR DESCRIPTION
**Pull request by dmpe**
In preparation of https://github.com/open-telemetry/opentelemetry-dotnet/issues/3259 and continuation of the previous 2 PRs, in this pull request, we deprecate use of B3 in API package. 

Instead of using OTEL.API nuget package, use now OTel.Extensions.Propagators nuget package for B3 tracing format

## Changes

* [x] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [x] Design discussion issue #1881 
* [ ] Changes in public API reviewed

### CODE REVIEW

**Comment by dmpe** [comment]
I am not sure if I am missing something, but methods have been annotated with `pragma` and hence it would be ready for a first look. Let me know if I forgot something. Thx.

Depending on the progress in https://github.com/open-telemetry/opentelemetry-dotnet/pull/3293, it will need to be rebased....

**Review by cijothomas — CHANGES_REQUESTED** [review]
Suggested improving obsolete message. This is 1st time we are Obsolete ing something from this repo, so need to make sure the migration is as smooth as possible.

**Comment by dmpe** [comment]
> Suggested improving obsolete message. This is 1st time we are Obsolete ing something from this repo, so need to make sure the migration is as smooth as possible.

Agree. I have slightly improved the wording of the deprecation message, plus added a comment to the class.

File: src/OpenTelemetry.Api/Context/Propagation/B3Propagator.cs [hunk 1]
```diff
@@ -25,7 +25,9 @@ namespace OpenTelemetry.Context.Propagation
 {
     /// <summary>
     /// A text map propagator for B3. See https://github.com/openzipkin/b3-propagation.
+    /// This class has been deprecated in favour of OpenTelemetry.Extensions.Propagators package.
     /// </summary>
+    [Obsolete("Use B3Propagator class from OpenTelemetry.Extensions.Propagators namespace, shipped as part of the same nuget package.")]
```
Comment (cijothomas) [inline src/OpenTelemetry.Api/Context/Propagation/B3Propagator.cs:30]:
```suggestion
    [Obsolete("Use B3Propagator class from OpenTelemetry.Extensions.Propagators namespace, shipped as part of OpenTelemetry.Extensions.Propagators package.")]
```

File: src/OpenTelemetry.Api/Context/Propagation/B3Propagator.cs [hunk 2]
```diff
@@ -25,7 +25,9 @@ namespace OpenTelemetry.Context.Propagation
 {
     /// <summary>
     /// A text map propagator for B3. See https://github.com/openzipkin/b3-propagation.
+    /// This class has been deprecated in favour of OpenTelemetry.Extensions.Propagators package.
     /// </summary>
+    [Obsolete("Use B3Propagator class from OpenTelemetry.Extensions.Propagators namespace, shipped as part of OpenTelemetry.Extensions.Propagators package.")]
```
Comment (cijothomas) [inline src/OpenTelemetry.Api/Context/Propagation/B3Propagator.cs:30]: This message looks good to me. @alanwest to offer any additional feedback.

We need to use the same message in all Obsolete places, and also in changelog.

Comment (dmpe) [reply src/OpenTelemetry.Api/Context/Propagation/B3Propagator.cs:30]: I have updated changelog and added the same message to all obsolete places.

**Review by alanwest — APPROVED** [review]
LGTM

**Review by cijothomas — APPROVED** [review]

**Pull request merged**